### PR TITLE
feat: inline assets in css file using `data:`

### PR DESCRIPTION
**Pull request by alan-agius4**
Closes: #263

## I'm submitting a...

```
[ ] Bug Fix
[X] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [X] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description
Inline assets such as fonts and images inside the css file.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by DavidParks8** [comment]
Readme documentation for the new field should probably be added.

**Comment by alan-agius4** [comment]
For sure, is there a readme which contains all the options that can be passed?

**Comment by DavidParks8** [comment]
For now, the best we’ve got is the main readme.

File: README.md [hunk 1]
```diff
@@ -168,6 +169,20 @@ For example, the following would use `index.ts` as the secondary entry point:
...
 
+#### How to embed assets in CSS?
+
+You can embed assets such as font and images inside the outputted css. More information [in the CSS tricks website](https://css-tricks.com/data-uris)
+
+```json
+{
+  "ngPackage": {
+    "lib": {
+      "embedAssets": true
```
Comment (dherges) [inline README.md]: Added new "config"

File: integration/samples/embed-assets/ng-package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "../../../src/ng-package.schema.json",
+  "lib": {
+    "entryFile": "public_api.ts",
+    "embedAssets": true,
```
Comment (dherges) [inline integration/samples/embed-assets/ng-package.json]: Integraton test added.

File: integration/samples/embed-assets/specs/bundle.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import * as fs from 'fs';
+import * as path from 'path';
+
+describe(`@sample/inline-assets`, () => {
+
+  describe(`bundle inline-assets.umd.js`, () => {
+    let BUNDLE;
+    before(() => {
+      BUNDLE = fs.readFileSync(
+        path.resolve(__dirname, '..', 'dist', 'bundles', 'inline-assets.umd.js'), 'utf-8');
```
Comment (dherges) [inline integration/samples/embed-assets/specs/bundle.ts]: `require()` the UMD bundle and/or the AoT metadata is better.

File: integration/samples/embed-assets/specs/bundle.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    });
+
+    it(`should exist`, () => {
+      expect(BUNDLE).to.be.ok;
+    });
+
+    it(`should have the image inlined with 'data:image'`, () => {
+      expect(BUNDLE).to.contain("data:image/png;base64");
```
Comment (dherges) [inline integration/samples/embed-assets/specs/bundle.ts]: See above. Replace with `METADATA['SomeComponent']['blablubb']['decorators'][0]['styles']`

File: src/lib/ng-packagr.ts [hunk 5]
```diff
@@ -43,6 +43,7 @@ export async function createNgPackage(opts: NgPackagrCliArguments): Promise<void
     }
 
     await copyFiles(`${rootPackage.sourcePath}/README.md`, rootPackage.destinationPath);
+    await copyFiles(`${rootPackage.sourcePath}/CHANGELOG.md`, rootPackage.destinationPath);
```
Comment (dherges) [inline src/lib/ng-packagr.ts]: New "feat: copies `CHANGELOG.md` to package destination"

File: src/ng-package.schema.json [hunk 6]
```diff
@@ -47,6 +47,11 @@
           "type": "string",
           "enum": ["preserve", "react", "react-native"],
           "default": ""
+        },
+        "embedAssets": {
```
Comment (dherges) [inline src/ng-package.schema.json]: Schema added! 👍

File: src/lib/ng-packagr.ts (hunk 5 above)
Comment (alan-agius4) [reply src/lib/ng-packagr.ts]: Absolutely right, i committed this by mistake :)

File: README.md (hunk 1 above)
Comment (alan-agius4) [reply README.md]: Can you please explain what you mean please?

**Comment by alan-agius4** [comment]
The postcss-url copy is also a cool option to add tbh. NG CLI and Webpack shouldn't encounter any issue with that.

**Comment by alan-agius4** [comment]
I thought about the ‘copy’ a bit more and actually now I see some problems, the main reason being that the CSS is the JS file, and thus webpack sass-loaders, css-loader won’t be able to parse the css and resolve the url of the image at application level.

I think as in general for library components, I’d rather try to avoid images, and if I really need them i’d go with embed images in order to avoid issues at consumer level.

**Comment by DavidParks8** [comment]
That’s what I ended up doing when I needed an image. It was embedded as an svg in its own component.

**Comment by alan-agius4** [comment]
I’d go as far as having ‘embedAssets’ option set to ‘true’ by default,
considering even that NG CLI is doing so. At least the last time I checked.

My only concern, that might be considered as a Breaking change, but since
there is a V2 coming up maybe it wouldn’t be such a problem.

**Comment by alan-agius4** [comment]
@dherges I am having some problems here with Travis, for some reason the new test is failing, however it does work locally and on circleci

**Comment by dherges** [comment]
It says `Error: Cannot find module '../dist/embed-assets.metadata.json'`

Can you rebase the changes on master and update the PR?

**Comment by alan-agius4** [comment]
I already did that, and seemed to work locally and on circle. but i can
give it another shot tomorrow.

**Comment by alan-agius4** [comment]
All green now :)

**Comment by alan-agius4** [comment]
Should we set this to true by default?

**Comment by dherges** [comment]
Hm. Maybe. First, let's change the property to an enum:

```
cssUrl: 'inline' | 'none'
```

**Comment by alan-agius4** [comment]
@dherges property changed

**Pull request merged**